api/src/main/java/org/hawkular/inventory/api/model/MetricUnit.java

```diff
@@ -26,13 +26,54 @@
  */
 public enum MetricUnit {
 
-
+    // Simple Metric Types - Absolute and Relative
     NONE(""),
-    MILLI_SECOND("ms"),
+    PERCENTAGE("%"),
+
+    // Absolute Sizes in Bytes (utilization)
+    BYTES("B"),
+    KILOBYTES("KB"),
+    MEGABYTES("MB"),
+    GIGABYTES("GB"),
+    TERABYTES("TB"),
+    PETABYTES("PB"),
+
+    // Absolute Sizes in Bits (throughput)
+    BITS("b"),
+    KILOBITS("Kb"),
+    MEGABITS("Mb"),
+    GIGABITS("Gb"),
+    TERABITS("Tb"),
+    PETABITS("Pb"),
+
+    // Absolute Time - no display, only hints to the UI how to display
+    EPOCH_MILLISECONDS(""),
+    EPOCH_SECONDS(""),
+
+    // Relative Time
+    JIFFYS("j"),
+    NANOSECONDS("ns"),
+    MICROSECONDS("us"),
+    MILLISECONDS("ms"),
     SECONDS("s"),
-    MINUTE("min"),
-    BYTE("b"),
-    KILO_BYTE("kb");
+    MINUTES("m"),
+    HOURS("h"),
+    DAYS("d"),
+
+    // Rate
+    PER_JIFFY("/j"),
```

Comment:
Jiffy is a strange thingy anyway :)

There is JSR 363 that may be helpful here.